mitmproxy/protocol/http.py

```diff
@@ -397,10 +400,17 @@ def authenticate(self, request):
             if self.config.authenticator.authenticate(request.headers):
                 self.config.authenticator.clean(request.headers)
             else:
-                self.send_response(models.make_error_response(
-                    407,
-                    "Proxy Authentication Required",
-                    http.Headers(**self.config.authenticator.auth_challenge_headers())
-                ))
+                if self.mode == "transparent":
+                    self.send_response(models.make_error_response(
+                        401,
+                        "Authentication Required",
+                        http.Headers(**self.config.authenticator.auth_challenge_headers())
+                    ))
+                else:
+                    self.send_response(models.make_error_response(
+                        407,
+                        "Proxy Authentication Required",
+                        http.Headers(**self.config.authenticator.auth_challenge_headers())
+                    ))
```

Comment:
If the only difference is the message, how do you feel about extracting it out to remove the branching?
```python
message = "Authentication Required" if self.mode == "transparent" else "Proxy Authentication Required"
self.send_response(models.make_error_response(
    407,
    message,
    http.Headers(**self.config.authenticator.auth_challenge_headers())
))
```